opentelemetry-exporter-cloud-trace/src/opentelemetry/tools/resource_detector.py

```diff
@@ -28,7 +35,34 @@ def get_gce_resources():
     return gce_resources
 
 
-_RESOURCE_FINDERS = [get_gce_resources]
+def get_gke_resources():
+    """ Resource finder for GKE attributes
+
+    """
+    # The user must specify these environment variables via the Downward API
+    container_name = os.getenv("CONTAINER_NAME")
+    pod_namespace = os.getenv("NAMESPACE")
+    if not container_name or not pod_namespace:
+        return {}
+    all_metadata = _get_all_google_metadata()
+    pod_name = os.getenv("HOSTNAME", "")
+    gke_resources = {
+        "cloud.account.id": all_metadata["project"]["projectId"],
+        "k8s.cluster.name": all_metadata["instance"]["attributes"][
+            "cluster-name"
+        ],
+        "k8s.namespace.name": pod_namespace,
+        "host.id": all_metadata["instance"]["id"],
+        "k8s.pod.name": pod_name,
+        "container.name": container_name,
+        "cloud.zone": all_metadata["instance"]["zone"].split("/")[-1],
+        "cloud.provider": "gcp",
```

Comment:
Is there any chance that the resource detector gets called while running on a non-google machine? (The GKE and GCP detectors assume we're already on Google, I'm just making sure that's a valid assumption.)

Reply:
I believe I discussed this with Chris and we agreed that if a user is calling the GoogleCloudResourceDetector they should be on a GCP tool. There is also an option to swallow that any errors of that nature by passing in `raise_on_error=False` to the detector.